File: BST/golang/tree.go
package main


type Tree struct {
	root *Node
}


func (self *Tree) Add(value int) bool {
	if self.root == nil {
		self.root = &Node{value: value, left:nil, right:nil}
		return true
	}
	return self.root.Add(value)
}

func (self *Tree) Contain(value int) bool {
	if self.root == nil {
		return false
	}
	return self.root.Contain(value)
}

func (self *Tree) Length() int {
	if self.root == nil {
		return 0
	}
	return self.root.Length()
}

func (self *Tree) Height() int {
	if self.root == nil {
		return 0
	}
	return self.root.Height()
}

File: BST/javascript/tree.js
const Node = require('./node.js');

class Tree {
    constructor() {
        this.root = null;
    }

    add(value) {
        if (this.root == null) {
            this.root = new Node(value);
            return true;
        }
        return this.root.add(value);
    }
    contain(value) {
        if (this.root == null) {
            return false;
        }
        return this.root.contain(value);
    }
    length() {
        return (this.root == null ? 0 : this.root.length())
    }
    height() {
        return (this.root == null ? 0 : this.root.height())
    }

}

module.exports = Tree;